pkg/queue/instance.go

```diff
@@ -87,7 +97,7 @@ func (q *queueImpl) get() (task Task, shutdown bool) {
 		q.cond.Wait()
 	}
 
-	if q.closing {
+	if q.closing && len(q.tasks) == 0 {
```

Comment:
This may be not right, we have places depending on: when the queue is closing, not process items anymore.  One place is deleting a cluster, stop processing services/endpoints for it.

Reply:
I can revert this

Reply:
fwiw its already broken if this is the desired behavior

```
	close(stop)
	q.Push(task)
```

can still process the task.

Reply:
Updated to revert this though